## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#012edf0c-19aa-4c15-9870-ddaf8d0d620b) |
| mint-gateway-s3.sh | :x: [more...](#3547f34d-bfa4-4447-b94d-ecd64701c546) |
| mint-gateway-azure.sh | :x: [more...](#e1742711-ff5d-40a3-9c98-b32371bb27dc) |
| mint-erasure.sh | :x: [more...](#ffd5b2af-2414-41fb-9023-f7f657a3adb3) |
| mint-dist-erasure.sh | :x: [more...](#e2a2aa9d-27d3-4a30-9cfb-56e01d423c66) |
| mint-zoned.sh | :x: [more...](#46028a67-81b2-496d-8030-ebcba2d02dce) |
<a name="ffd5b2af-2414-41fb-9023-f7f657a3adb3"></a>
### 10468-d0eeff2/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:32199
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2925177fd4dd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 9 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 36 seconds
(8/15) Running minio-dotnet tests ... done in 32 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 38 seconds
(10/15) Running minio-java tests ... done in 49 seconds
(11/15) Running minio-js tests ... FAILED in 21 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 7,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... FAILED in 1 seconds
{
  "name": "minio-py:test_copy_object_no_copy_condition",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1b9cc0a6-c4c2-4780-83b4-90e01445fbc7",
    "object_source": "7fcf4c19-18a1-4d58-a03e-fbe385b073cb-source",
    "object_name": "7fcf4c19-18a1-4d58-a03e-fbe385b073cb-copy"
  },
  "message": "version-id mismatch. expected=None, got=null",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 485, in test_copy_object_no_copy_condition\n    _validate_stat(st_obj, size, {})\n  File \"/mint/run/core/minio-py/tests.py\", line 432, in _validate_stat\n    version_id, st_obj.version_id,\nValueError: version-id mismatch. expected=None, got=null\n",
  "duration": 98
}
(11/15) Running s3cmd tests ... done in 22 seconds
(12/15) Running s3select tests ... done in 4 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="e2a2aa9d-27d3-4a30-9cfb-56e01d423c66"></a>
### 10468-d0eeff2/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32670
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 70d58882baf9:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 15 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 3 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 0 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 28 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 17 seconds
(11/15) Running minio-js tests ... FAILED in 30 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 7,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... FAILED in 3 seconds
{
  "name": "minio-py:test_copy_object_no_copy_condition",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-20baefec-141e-4d6b-970e-3bc69f03fa1f",
    "object_source": "6e648853-12a8-4fb3-b0c7-ff00ecf915f9-source",
    "object_name": "6e648853-12a8-4fb3-b0c7-ff00ecf915f9-copy"
  },
  "message": "version-id mismatch. expected=None, got=null",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 485, in test_copy_object_no_copy_condition\n    _validate_stat(st_obj, size, {})\n  File \"/mint/run/core/minio-py/tests.py\", line 432, in _validate_stat\n    version_id, st_obj.version_id,\nValueError: version-id mismatch. expected=None, got=null\n",
  "duration": 278
}
(11/15) Running s3cmd tests ... done in 38 seconds
(12/15) Running s3select tests ... done in 7 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="012edf0c-19aa-4c15-9870-ddaf8d0d620b"></a>
### 10468-d0eeff2/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30551
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e06349d896e2:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 2 minutes and 11 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 59 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 4 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 27 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 33 seconds
(11/15) Running minio-js tests ... FAILED in 22 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 10,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... FAILED in 3 seconds
{
  "name": "minio-py:test_copy_object_no_copy_condition",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-d3bd4463-2301-40a4-b39c-56120c101aa0",
    "object_source": "064e867d-3a61-4966-ba9d-a9ce9d9ad870-source",
    "object_name": "064e867d-3a61-4966-ba9d-a9ce9d9ad870-copy"
  },
  "message": "version-id mismatch. expected=None, got=null",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 485, in test_copy_object_no_copy_condition\n    _validate_stat(st_obj, size, {})\n  File \"/mint/run/core/minio-py/tests.py\", line 432, in _validate_stat\n    version_id, st_obj.version_id,\nValueError: version-id mismatch. expected=None, got=null\n",
  "duration": 384
}
(11/15) Running s3cmd tests ... done in 36 seconds
(12/15) Running s3select tests ... done in 8 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="3547f34d-bfa4-4447-b94d-ecd64701c546"></a>
### 10468-d0eeff2/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32682
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 48c10f36e986:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 11 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 35 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 29 seconds
(10/15) Running minio-java tests ... done in 55 seconds
(11/15) Running minio-js tests ... FAILED in 19 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... FAILED in 2 seconds
{
  "name": "minio-py:test_copy_object_etag_match",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-05ad7669-2430-40fd-bead-e0e5c991720c",
    "object_source": "8933e211-30ce-486a-aecc-a42758ab2f7c-source",
    "object_name": "8933e211-30ce-486a-aecc-a42758ab2f7c-copy",
    "conditions": {
      "set_match_etag": "7defa763ba19b52e236202b5c5abb3e5"
    }
  },
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 566, in test_copy_object_etag_match\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 413, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2044, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "duration": 256
}
(11/15) Running s3cmd tests ... done in 1 minutes and 9 seconds
(12/15) Running s3select tests ... done in 4 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="e1742711-ff5d-40a3-9c98-b32371bb27dc"></a>
### 10468-d0eeff2/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31288
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 63fc4a36ba3c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 14 seconds
(4/15) Running aws-sdk-ruby tests ... done in 17 seconds
(5/15) Running awscli tests ... FAILED in 43 seconds
{
  "name": "awscli",
  "duration": 4088,
  "function": "aws --endpoint-url http://minio-dev6.minio.io:31288 s3api copy-object --bucket awscli-mint-test-bucket-1012 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-1012/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected a861ac626e0bee4421a50737c581233c, got 0x8D856A18743D0BE-1"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 4 minutes and 23 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 33 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 50 seconds
{
  "args": {
    "bucketName": "minio-go-test-6vws9prdnduknkqx",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 718,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 6 minutes and 58 seconds
(9/15) Running minio-js tests ... FAILED in 47 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 80,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(9/15) Running minio-py tests ... FAILED in 12 minutes and 36 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-234ee858-5acb-4122-b494-7d221a1bac24",
    "object_name": "5fe6ef14-967a-4ea1-980f-94d7380f82bf"
  },
  "message": "InvalidSizeError: message: Data written 6291456 bytes is smaller than the specified size 11534336 bytes",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1667, in test_thread_safe\n    raise exceptions[0]\n  File \"/mint/run/core/minio-py/tests.py\", line 1627, in get_object_and_check\n    _CLIENT.fget_object(bucket_name, object_name, local_file)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 822, in fget_object\n    raise InvalidSizeError(msg)\nminio.error.InvalidSizeError: InvalidSizeError: message: Data written 6291456 bytes is smaller than the specified size 11534336 bytes\n",
  "duration": 35482
}
(9/15) Running s3cmd tests ... done in 2 minutes and 25 seconds
(10/15) Running s3select tests ... done in 35 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
<a name="46028a67-81b2-496d-8030-ebcba2d02dce"></a>
### 10468-d0eeff2/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31120
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 73a0aa29c1cb:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 21 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 48 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 3 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 49 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 25 seconds
(11/15) Running minio-js tests ... FAILED in 21 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(11/15) Running minio-py tests ... FAILED in 2 seconds
{
  "name": "minio-py:test_copy_object_no_copy_condition",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-d3d77b6d-1a26-4f44-92ab-9750097c19e4",
    "object_source": "92842cfb-1801-4ded-91ce-298b6b5cf284-source",
    "object_name": "92842cfb-1801-4ded-91ce-298b6b5cf284-copy"
  },
  "message": "version-id mismatch. expected=None, got=null",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 485, in test_copy_object_no_copy_condition\n    _validate_stat(st_obj, size, {})\n  File \"/mint/run/core/minio-py/tests.py\", line 432, in _validate_stat\n    version_id, st_obj.version_id,\nValueError: version-id mismatch. expected=None, got=null\n",
  "duration": 399
}
(11/15) Running s3cmd tests ... done in 27 seconds
(12/15) Running s3select tests ... done in 14 seconds
(13/15) Running security tests ... done in 1 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10468-d0eeff2

--- a/cmd/signals.go
+++ b/cmd/signals.go
@@ -18,6 +18,8 @@ package cmd
 
 import (
 	"context"
+	"errors"
+	"net/http"
 	"os"
 	"strings"
 
@@ -53,7 +55,9 @@ func handleSignals() {
 
 		if httpServer := newHTTPServerFn(); httpServer != nil {
 			err = httpServer.Shutdown()
-			logger.LogIf(context.Background(), err)
+			if !errors.Is(err, http.ErrServerClosed) {
+				logger.LogIf(context.Background(), err)
+			}
 		}
 
 		if objAPI := newObjectLayerWithoutSafeModeFn(); objAPI != nil {

--- a/cmd/sts-handlers.go
+++ b/cmd/sts-handlers.go
@@ -88,7 +88,7 @@ func registerSTSRouter(router *mux.Router) {
 		ctypeOk := wildcard.MatchSimple("application/x-www-form-urlencoded*", r.Header.Get(xhttp.ContentType))
 		noQueries := len(r.URL.Query()) == 0
 		return ctypeOk && noQueries
-	}).HandlerFunc(httpTraceAll(sts.AssumeRoleWithJWT))
+	}).HandlerFunc(httpTraceAll(sts.AssumeRoleWithSSO))
 
 	// AssumeRoleWithClientGrants
 	stsRouter.Methods(http.MethodPost).HandlerFunc(httpTraceAll(sts.AssumeRoleWithClientGrants)).
@@ -258,8 +258,8 @@ func (sts *stsAPIHandlers) AssumeRole(w http.ResponseWriter, r *http.Request) {
 	writeSuccessResponseXML(w, encodeResponse(assumeRoleResponse))
 }
 
-func (sts *stsAPIHandlers) AssumeRoleWithJWT(w http.ResponseWriter, r *http.Request) {
-	ctx := newContext(r, w, "AssumeRoleJWTCommon")
+func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "AssumeRoleSSOCommon")
 
 	// Parse the incoming form data.
 	if err := r.ParseForm(); err != nil {
@@ -274,6 +274,9 @@ func (sts *stsAPIHandlers) AssumeRoleWithJWT(w http.ResponseWriter, r *http.Requ
 
 	action := r.Form.Get(stsAction)
 	switch action {
+	case ldapIdentity:
+		sts.AssumeRoleWithLDAPIdentity(w, r)
+		return
 	case clientGrants, webIdentity:
 	default:
 		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("Unsupported action %s", action))
@@ -417,7 +420,7 @@ func (sts *stsAPIHandlers) AssumeRoleWithJWT(w http.ResponseWriter, r *http.Requ
 // Eg:-
 //    $ curl https://minio:9000/?Action=AssumeRoleWithWebIdentity&WebIdentityToken=<jwt>
 func (sts *stsAPIHandlers) AssumeRoleWithWebIdentity(w http.ResponseWriter, r *http.Request) {
-	sts.AssumeRoleWithJWT(w, r)
+	sts.AssumeRoleWithSSO(w, r)
 }
 
 // AssumeRoleWithClientGrants - implementation of AWS STS extension API supporting
@@ -426,7 +429,7 @@ func (sts *stsAPIHandlers) AssumeRoleWithWebIdentity(w http.ResponseWriter, r *h
 // Eg:-
 //    $ curl https://minio:9000/?Action=AssumeRoleWithClientGrants&Token=<jwt>
 func (sts *stsAPIHandlers) AssumeRoleWithClientGrants(w http.ResponseWriter, r *http.Request) {
-	sts.AssumeRoleWithJWT(w, r)
+	sts.AssumeRoleWithSSO(w, r)
 }
 
 // AssumeRoleWithLDAPIdentity - implements user auth against LDAP server

